### PR TITLE
Prevent any record_soft_failure which does not link to any ticket reference

### PR DESCRIPTION
**Pull request by okurz**
Our convention is to only use record_soft_failure in combination with a valid
bug reference, a ticket reference or also a fate link to prevent soft-failing
test cases which no one cares to make passed because there is simply no ticket
in any other issue management tool. This commit introduces a new check to
prevent inclusion of any new record_soft_failure without these valid
references.

### CODE REVIEW

**Comment by okurz** [comment]
@asmorodskyi you could help me so that this PR also does not need to stay open for more than a year ;)

File: tests/sles4sap/sapconf.pm [hunk 1]
```diff
@@ -43,34 +43,34 @@ sub run_developers_tests {
...
     if ($output !~ /sapconf_test\.sh/) {
-        record_soft_failure 'Script sapconf_test.sh is not in the developer test package';
+        record_info 'Script not found', 'Script sapconf_test.sh is not in the developer test package';
         return;
     }
     assert_script_run "chmod +x sapconf_test.sh";
     $ret = script_run "./sapconf_test.sh -c local -p no | tee $log", 600;
     # Record soft fail only if script returns an error. Ignore timeout as test completion is checked below
-    record_soft_failure "sapconf_test.sh returned error code: [$ret]" if (defined $ret and $ret != 0);
+    record_info('Test failed', "sapconf_test.sh returned error code: [$ret]", result => 'fail') if (defined $ret and $ret != 0);
```
Comment (alvarocarvajald) [inline tests/sles4sap/sapconf.pm]: Can this `result => 'fail'` be changed for a `result => 'softfail'`? I do not consider an error code from the execution of the provided test shell script reason enough to assume there is a problem with sapconf, but with the test script itself.

File: tests/sles4sap/sapconf.pm [hunk 2]
```diff
@@ -43,34 +43,34 @@ sub run_developers_tests {
...
     upload_logs $log;
 
     # Check summary of tests on log for bug report
     my $report = script_output "grep ^Test $log || true";
-    record_soft_failure 'No tests summaries in log' unless ($report);
+    record_info('No tests summaries in log', result => 'fail') unless ($report);
```
Comment (alvarocarvajald) [inline tests/sles4sap/sapconf.pm]: Same as L68, please change `fail` for `softfail`.

File: tests/sles4sap/sapconf.pm [hunk 3]
```diff
@@ -79,7 +79,7 @@ sub run_developers_tests {
             record_soft_failure "bsc#$1";
         }
         else {
-            record_soft_failure $summary;
+            record_info $summary. result => 'fail';
```
Comment (alvarocarvajald) [inline tests/sles4sap/sapconf.pm]: That dot `.` should be a colon `,`, right? Or I am going blind and it is already a colon?

Also, the test shell script is now referencing 3 tests that are not tied to a bsc (example in: https://openqa.suse.de/tests/1750092#step/sapconf/111), one of those is tied to a fate, so I'd like to add this block before the else:

```
elsif ($summary =~ /Test #fate([0-9]+)/) { record_soft_failure "fate#$1"; }
```

Or better yet, re-write the regexp on the if as so:

```
if ($summary =~ /Test #(bsc|fate)([0-9]+)/) {
    record_soft_failure $1 . "#" . $2;
}
```

File: lib/utils.pm [hunk 4]
```diff
@@ -867,7 +867,7 @@ sub addon_license {
         do {
             assert_screen \@tags;
             if (match_has_tag('import-untrusted-gpg-key')) {
-                record_soft_failure 'untrusted gpg key';
+                record_info 'untrusted gpg key', "Trusting untrusted GPG key", result => 'softfail';
```
Comment (asmorodskyi) [inline lib/utils.pm]: why you not 'die' here ?

**Comment by asmorodskyi** [comment]
LGTM , please solve what @alvarocarvajald  asks and you can merge

**Pull request merged**